Avoid mode switching if the current mode already matches the requested one.

This works around a bug in OS X where CGDisplayCopyAllDisplayModes
does not return all available modes, and in fact sometimes does not
even include the currently active mode. This breaks the mode
restoration after full screen, as introduced by OpenJDK commit
http://hg.openjdk.java.net/jdk7u/jdk7u/jdk/rev/bf5bf37108c1. To avoid
the problem, we compare the requested and the actual display mode. If
they match, no switching is required.

As a drive-by fix, we also avoid a comparison between double and int.

First report: http://java.net/jira/browse/MACOSX_PORT-805
Oracle internal review ID: 2371768
Public Oracle bug ID: unknown, apparently not assigned yet

Index: jdk/src/macosx/classes/sun/awt/CGraphicsDevice.java
===================================================================
--- jdk/src/macosx/native/sun/awt/CGraphicsDevice.m	Thu Jan 16 11:52:13 2014 -0800
+++ jdk/src/macosx/native/sun/awt/CGraphicsDevice.m	Fri Jan 17 14:36:07 2014 +0100
@@ -76,7 +76,7 @@
 
         // Refresh rate might be 0 in display mode and we ask for specific display rate
         // but if we do not find exact match then 0 refresh rate might be just Ok
-        if (CGDisplayModeGetRefreshRate(cRef) == refrate) {
+        if ((int) CGDisplayModeGetRefreshRate(cRef) == refrate) {
             // Exact match
             return cRef;
         }
